python/pyarrow/_dataset.pyx

```diff
@@ -1416,6 +1435,11 @@ cdef class DirectoryPartitioning(Partitioning):
     ----------
     schema : Schema
         The schema that describes the partitions present in the file path.
+    dictionaries : List[Array]
+        If the type of any field of `schema` is a dictionary type, the
+        corresponding entry of `dictionaries` must be an array containing
+        every value which may be taken by the corresponding column or an
+        error will be raised in parsing.
```

Comment:
This seems a bit weird and inconvenient as an API. Why not accept a `Dict[str, Array]` mapping field names to dictionaries?